batavia/VirtualMachine.js

```diff
@@ -1818,44 +1920,82 @@ VirtualMachine.prototype.byte_CALL_FUNCTION_VAR = function(arg) {
+    }
     var kwargs = this.pop()
     return this.call_function(arg, null, kwargs)
 }
 
 VirtualMachine.prototype.byte_CALL_FUNCTION_VAR_KW = function(arg) {
-    var items = this.popn(2)
-    return this.call_function(arg, items[0], items[1])
+    if (constants.BATAVIA_MAGIC === constants.BATAVIA_MAGIC_36) {
+        // opcode: CALL_FUNCTION_EX
+        var kwargs
+        if (arg & 1) {
+            kwargs = this.pop()
+        }
+        var args = this.pop()
+        return this.call_function(0, args, kwargs)
+    } else {
+        var items = this.popn(2)
+        return this.call_function(arg, items[0], items[1])
+    }
 }
 
 VirtualMachine.prototype.call_function = function(arg, args, kwargs) {
-    // @arg is based on
-    // https://docs.python.org/3/library/dis.html#opcode-CALL_FUNCTION
-    var lenKw = Math.floor(arg / 256)
-    var lenPos = arg % 256
-    var namedargs = new types.JSDict()
-    for (var i = 0; i < lenKw; i++) {
-        var items = this.popn(2)
-        namedargs[items[0]] = items[1]
-    }
-    if (kwargs) {
-        for (let kv of kwargs.items()) {
-            namedargs[kv[0]] = kv[1]
+    if (constants.BATAVIA_MAGIC === constants.BATAVIA_MAGIC_36) {
+        var namedargs = new types.JSDict()
+        var lenPos = arg
+        if (kwargs) {
+            for (let kv of kwargs.items()) {
+                namedargs[kv[0]] = kv[1]
+            }
         }
-    }
-    var posargs = this.popn(lenPos)
-    if (args) {
-        for (let elem of args) {
-            posargs.push(elem)
+        var posargs = this.popn(lenPos)
+        if (args) {
+            for (let elem of args) {
+                posargs.push(elem)
+            }
+        }
+        var func = this.pop()
+        if (func.__call__ !== undefined) {
+            func = func.__call__.bind(func)
         }
-    }
 
-    var func = this.pop()
-    if (func.__call__ !== undefined) {
-        func = func.__call__.bind(func)
-    }
+        var retval = func(posargs, namedargs)
+        this.push(retval)
+    } else {
+        var namedargs = new types.JSDict()
+        var lenKw = Math.floor(arg / 256)
+        var lenPos = arg % 256
+        for (var i = 0; i < lenKw; i++) {
+            var items = this.popn(2)
+            namedargs[items[0]] = items[1]
+        }
+        if (kwargs) {
+            for (let kv of kwargs.items()) {
+                namedargs[kv[0]] = kv[1]
+            }
+        }
+        var posargs = this.popn(lenPos)
+        if (args) {
+            for (let elem of args) {
+                posargs.push(elem)
+            }
+        }
+        var func = this.pop()
+        if (func.__call__ !== undefined) {
+            func = func.__call__.bind(func)
+        }
 
-    var retval = func(posargs, namedargs)
-    this.push(retval)
+        var retval = func(posargs, namedargs)
+        this.push(retval)
+    }
 }
 
 VirtualMachine.prototype.byte_RETURN_VALUE = function() {
```

Comment:
At column 13: [(no-redeclare) 'retval' is already defined.](http://eslint.org/docs/rules/no-redeclare)